From f65b4e0161fe87bd6ddcb9c633c7b7bf52584c0a Mon Sep 17 00:00:00 2001
From: xhe <xw897002528@gmail.com>
Date: Sun, 16 Dec 2018 15:42:25 +0800
Subject: [PATCH 23/48] poparser: return if the buf is too small if line_len,
 e.g. inputbytes_left is not 0, that means the output buf is large enough.

---
 src/poparser.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/poparser.c b/src/poparser.c
index e71e386..874e835 100644
--- a/src/poparser.c
+++ b/src/poparser.c
@@ -165,6 +165,9 @@ enum po_error poparser_feed_line(struct po_parser *p, char* in, size_t in_len) {
 			if ( (len = iconv(p->cd, &line, &line_len, &x, &len)) == (size_t)-1) 
 				return -po_failed_iconv;
 
+			if (line_len != 0)
+				return -po_failed_iconv;
+
 			x = p->buf;
 		}
 
@@ -225,6 +228,9 @@ enum po_error poparser_feed_line(struct po_parser *p, char* in, size_t in_len) {
 			if ( (len = iconv(p->cd, &line, &line_len, &x, &len)) == (size_t)-1) 
 				return -po_failed_iconv;
 
+			if (line_len != 0)
+				return -po_failed_iconv;
+
 			x = p->buf;
 		}
 
-- 
2.25.0

